lib/data/repositories/collection_purchase_repository.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:memo/data/gateways/purchase_gateway.dart';
+import 'package:memo/data/gateways/sembast_database.dart';
+import 'package:memo/data/repositories/collection_repository.dart';
+import 'package:memo/data/serializers/collection_serializer.dart';
+
+abstract class CollectionPurchaseRepository {
+  /// Purchase products in the app with the store ID [storeId] for the local user.
+  Future<void> purchaseInApp({required String storeId});
+
+  /// Receives purchase information made by the user.
+  Future<List<String>> getPurchasesInfo();
+
+  /// Check which products are available for purchase.
+  Future<List<String>> isAvailable();
+
+  /// Updates the collection with the [id] to be premium or not.
+  Future<void> updatePurchaseCollection({required String id, required bool isPremium});
```

Comment:
Por exemplo esse método n precisaria estar aqui.

A n ser q fosse atualizar um produto como comprado em outro banco e assim tu ter essa informação de se algo está disponível para ser usado, essa seria a abordagem mais correta, porém teria q alterar bastante coisa, então vamos só tentar ver um lugar melhor pra esse método no momento.

Independente de pra onde esse método for, colocar um TODO nele, de q devemos melhorar esse fluxo.